processing/src/main/java/io/druid/query/topn/TopNQueryEngine.java

```diff
@@ -124,4 +129,13 @@ public TopNQueryEngine(StupidPool<ByteBuffer> bufferPool)
 
     return new TopNMapFn(query, topNAlgorithm);
   }
+
+  public static Boolean useOptimized(TopNQuery query)
```

Comment:
can we give this function a more descriptive name, such as `canApplyExtractionInPost` to make it less confusing as to what this means?